commonutils/src/main/java/io/intrepid/commonutils/IntrepidJavaUtils.java

```diff
@@ -0,0 +1,59 @@
+package io.intrepid.commonutils;
+
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import java.util.Collections;
+import java.util.List;
+
+public class IntrepidJavaUtils {
+    private final static char[] hexArray = "0123456789abcdef".toCharArray();
+
+    @NonNull
+    public static String bytesToHexString(@NonNull byte[] bytes) {
+        char[] hexChars = new char[bytes.length * 2];
+        for (int j = 0; j < bytes.length; j++) {
+            int v = bytes[j] & 0xFF;
+            hexChars[j * 2] = hexArray[v >>> 4];
+            hexChars[j * 2 + 1] = hexArray[v & 0x0F];
```

Comment:
It wasn't for the performance reasons, I just haven't thought about using String.format(). I'm fine with going with that approach if you think it's more readable.